Issue 5291 - Harden ReplicationManager.wait_for_replication

progier389 opened the pull request:
ReplicationManager.wait_for_replication may wrongly fails without no way to determine if the error is normal or not.
Solution to harden the function is to also check the description value ob the "from" side
And if servers are in sync but not having the expected value

LOG A DIFFERENT MESSAGE
REDO THE CHANGE (because previous one may have been overwritten by URP)

If the timeout (now set to 60 seconds by default) expires then log last 30 lines from replication plugin in error log
 on both instances.  (to capture the last replication session)
This change helped me to understand why test_change_repl_passwd[using-bind-group] fails while it pass when running alone.
 So I also fixed test_fetch_bindDnGroup test to restore the configuration and avoid impacting the other tests.

Issue 5291

Reviewed by: @mreynolds389

progier389 committed:
Issue 5291 - Harden ReplicationManager.wait_for_replication

ReplicationManager.wait_for_replication may wrongly fails without no way to determine if the error is normal or not.
Solution to harden the function is to also check the description value ob the "from" side
And if servers are in sync but not having the expected value
  - LOG A DIFFERENT MESSAGE
  - REDO THE CHANGE (because previous one may have been overwritten by URP)
Also logis the replication plugin logs if replication fails

Issue 5291

Reviewed by: ?

mreynolds389 reviewed (approved):
Besides my request about changing the default timeout the rest looks great!

progier389 committed:
Issue 5291 - Harden ReplicationManager.wait_for_replication

ReplicationManager.wait_for_replication may wrongly fails without no way to determine if the error is normal or not.
Solution to harden the function is to also check the description value ob the "from" side
And if servers are in sync but not having the expected value

LOG A DIFFERENT MESSAGE
REDO THE CHANGE (because previous one may have been overwritten by URP)

If the timeout (now set to 60 seconds by default) expires then log last 30 lines from replication plugin in error log
 on both instances.  (to capture the last replication session)
This change helped me to understand why test_change_repl_passwd[using-bind-group] fails while it pass when running alone.
 So I also fixed test_fetch_bindDnGroup test to restore the configuration and avoid impacting the other tests.

Issue 5291

Reviewed by: @mreynolds
